src/nvim/option.c

```diff
@@ -3345,7 +3332,7 @@ static char *option_get_valid_types(OptIndex opt_idx)
 /// @return  True if option is hidden, false otherwise. Returns false if option name is invalid.
 bool is_option_hidden(OptIndex opt_idx)
 {
-  return opt_idx == kOptInvalid ? false : get_varp(&options[opt_idx]) == NULL;
+  return opt_idx == kOptInvalid ? false : options[opt_idx].hidden;
```

Comment:
`exists('+option')` should probably return `false` for `immutable` option as well, but may that can be done in another PR.

Reply:
Should it? The way I see it, `immutable` options are options that do exist, but just cannot have their value changed. The way I implemented it, options could be set to `immutable` temporarily as well, if we ever want them to not be changed.

Reply:
Hmm, I don't think this matters much for now.